app/components/AddDelegateModal/index.js

```diff
@@ -81,15 +56,35 @@ class AddDelegateModal extends Component<Props> {
     );
   };
 
-  renderCreationBody = () => {
+  createAccount = async () =>  {
```

Comment:
I think since we are using redux thunks it's better to keep all the async code [in the thunk](https://www.javascriptstuff.com/react-ajax-best-practices/). In that case, you would have to have your loading flag in the store as well. I think it makes it easier for people to reason with but it's just an opinion so feel free to disregard ¯\_(ツ)_/¯